### DISCUSSION TITLE
Reading a CSV file with an empty header is problematic

### DISCUSSION THREAD

**Discussion by szarnyasg** [duckdb/duckdb#4932] [General]
### What happens?

When loading a CSV file with an empty header, the first row with actual content is either skipped (with `HEADER true`) or it is read but its first attribute is not loaded correctly (with `HEADER false`).

Obviously, such a CSV file is malformed so it's unclear how it should be loaded, so one can argue this is not a bug. That said, IMHO loading half the first row violates the principle of least surprise -- either throwing an error message or loading the full row would be better alternatives.

### To Reproduce

Create `my.csv` with an empty first line:
```csv

a 1
b 2
c 3
```
Run the following commands in DuckDB:
```sql
CREATE OR REPLACE TABLE stats (file varchar, numEntities bigint);
COPY stats FROM 'my.csv' (DELIMITER ' ');
SELECT * FROM stats;
```
```console
┌──────┬─────────────┐
│ file │ numEntities │
├──────┼─────────────┤
│      │ 1           │
│ b    │ 2           │
│ c    │ 3           │
└──────┴─────────────┘
```
Note that the `file` attribute is empty for the first row.

With the `HEADER` setting, the line (`a 1`) is skipped in the CSV file:
```sql
CREATE OR REPLACE TABLE stats (file varchar, numEntities bigint);
COPY stats FROM 'my.csv' (DELIMITER ' ', HEADER);
SELECT * FROM stats;
```
```console
┌──────┬─────────────┐
│ file │ numEntities │
├──────┼─────────────┤
│ b    │ 2           │
│ c    │ 3           │
└──────┴─────────────┘
```

### OS:

MacOS

### DuckDB Version:

0.5.1

### DuckDB Client:

CLI

### Full Name:

Gabor Szarnyas

### Affiliation:

CWI

### Have you tried this on the latest `master` branch?

- [X] I agree

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] I agree

**Comment by hannes** [comment]
I get that this is annoying. We could consider skipping entirely empty lines, but those could be valid in the case of a single-column file...
